IntegrationTests/Services/AWSSTSIntegrationTests/STSAssumeRoleAWSCredentialIdentityResolverTests.swift

```diff
@@ -0,0 +1,130 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import XCTest
+import AWSSTS
+import AWSIAM
+import AWSSDKIdentity
+import ClientRuntime
+
+class STSAssumeRoleAWSCredentialIdentityResolverTests: XCTestCase {
+    private let region = "us-east-1"
+
+    // MARK: - Clients, policies, and variables
+
+    // STS client with only the STSAssumeRoleAWSCredentialIdentityResolver configured.
+    private var assumeRoleStsClient: STSClient!
+    private var assumeRoleStsConfig: STSClient.STSClientConfiguration!
+
+    // Used to create temporary role assumed by STS assume role credentials provider.
+    private var iamClient: IAMClient!
+    private let roleName = "aws-sts-integration-test-\(UUID().uuidString.split(separator: "-").first!.lowercased())"
+    private let roleSessionName = "aws-sts-integration-test-\(UUID().uuidString.split(separator: "-").first!.lowercased())"
+    private var roleArn: String!
+
+    // JSON assume role policy
+    private func assumeRolePolicy(acctID: String) -> String {
+        return """
+        {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Principal": {
+                        "AWS": "arn:aws:iam::\(acctID):root"
+                    },
+                    "Action": "sts:AssumeRole"
+                }
+            ]
+        }
+        """
+    }
+
+    private let identityPolicyName = "temp-allow-STS-getCallerIdentity"
+    // JSON identity policy
+    private let roleIdentityPolicy = """
+    {"Version": "2012-10-17","Statement": [{"Sid": "","Effect": "Allow",
+    "Action": "sts:GetCallerIdentity","Resource": "*"}]}
```

Comment:
nit: consider creating a constant and moving these to another file

Reply:
For now this identity policy is only used by this test, so it makes sense to have it in same file as the tests for cohesion. When new integration tests down the line use identical policies, then we can consider refactoring it.